### ISSUE TITLE
Using TabbedPanel's switch_to does not cause the scroll view for the header strip to update if the tab is off the screen

### ISSUE THREAD

**Issue by bluelight773** [kivy/kivy#3645]
When I use switch_to to switch tabs programmatically, the expected tab's contents do show, but if the tab happens to be off the screen (because of many tabs open at once) then the scroller doesn't update as to ensure the switched-to tab is shown.

**Comment by udiboy1209** [comment]
I would like to solve this issue.

Plus @bluelight773 could you provide the code you used which would reproduce this bug?

**Comment by udiboy1209** [comment]
Could not replicate :confused: 
Tried [this code](http://paste.ubuntu.com/14257186/) on Ubuntu 14.04 with latest Kivy source build

**Comment by andong777** [comment]
I reproduced it using the code below. I think what @bluelight773 described is, if you have many tabs in your app and you switch to a tab in code by clicking the button, the switched-to tab is highlighted but not shown. Try running my code, scroll to the rightmost tab, and click the button.
I tried to solve this issue but encountered a problem. I want to use the `scroll_to` function of ScrollView in `switch_to` function of TabbedPanel, but it seems the scrollview of the tabbed panel doesn't have a parent. And if you add the scrollview to the panel, it will show that

```
kivy.uix.widget.WidgetException: Cannot add <kivy.uix.scrollview.ScrollView object at 0x1118c7d48>, it already has a parent <kivy.uix.tabbedpanel.TabbedPanelContent object at 0x1118c7250>
```

I also tried setting `scroll_x`. It works but the code is not as concise as calling `scroll_to`, and there is no animation.
@udiboy1209 you can continue to solve this issue, or give me a hint and let me do it.


```
from kivy.app import App
from kivy.uix.tabbedpanel import TabbedPanel, TabbedPanelItem
from kivy.uix.boxlayout import BoxLayout
from kivy.uix.button import Button
from kivy.uix.label import Label


class TabbedPanelApp(App):

    def click(self, touch):
        self.panel.switch_to(self.default)

    def build(self):
        box = BoxLayout(orientation='vertical')
        button = Button(text='a button', on_press=self.click)
        self.panel = TabbedPanel(do_default_tab=False)
        self.default = TabbedPanelItem(text='default tab')
        label = Label(text='content of default tab')
        self.default.add_widget(label)
        self.panel.add_widget(self.default)
        for i in range(10):
            item = TabbedPanelItem(text='tab ' + str(i))
            label = Label(text='content of tab ' + str(i))
            item.add_widget(label)
            self.panel.add_widget(item)

        box.add_widget(button)
        box.add_widget(self.panel)
        return box

if __name__ == '__main__':
    TabbedPanelApp().run()
```

**Comment by udiboy1209** [comment]
@andong777 your example works for me too :confused:. I have Tab 9 selected and scrolled out of view, but the button click still brings up "Content of default tab" correctly. 

Maybe its a env related issue. I'm running kivy master on Ubuntu 14.04, with SDL2. @bluelight773 and @andong777 post your environment details so they can be compared.

Plus, I'd appreciate if you take up this issue. Planning to work on another one.

**Comment by andong777** [comment]
@udiboy1209 Yes, the content of default tab is brought up, but is 'default tab' also shown on your computer? If I scroll to tab 9 and click the button, the content is replaced but the default tab is still out of window.
Actually, I'm not sure if it needs a fix, maybe it is just a feature. How do you think about it?
I run kivy 1.9.1 on OS X 10.11, Python 3.5 and SDL2.

Demo: After the button click, 'default tab' is not shown.
![image](https://cloud.githubusercontent.com/assets/3743928/14736141/ab74fc28-08a7-11e6-8db3-8dcf6453f4ff.png)

**Comment by udiboy1209** [comment]
Oh! It doesn't scroll to "Default tab" which is probably the missing thing here, but it does get highlighted.
